addons/stock/models/stock_move.py

```diff
@@ -1108,6 +1108,16 @@ def _action_confirm(self, merge=True, merge_into=False):
             moves = self._merge_moves(merge_into=merge_into)
         # call `_action_assign` on every confirmed move which location_id bypasses the reservation
         moves.filtered(lambda move: not move.picking_id.immediate_transfer and move._should_bypass_reservation() and move.state == 'confirmed')._action_assign()
+        for m in move_to_confirm:
+            picking_type = None
+            try:
```

Comment:
Why do you need a try catch?

Reply:
because not all moves have a picking type or are linked to a picking (having a picking type)
: highlighted by the tests giving a lot of exceptions

Reply:
You probably just need.
```python3
picking_type = m.picking_type_id
if not picking_type and m.picking_id:
    picking_type = m.picking_id.picking_type_id
```
Because in your case, m.picking_id return False and False.picking_type_id is returning the error. Just check if the field is set.

You can use this. Smaller but it will take the picking's picking type first but it should not be an issue since they should be the same
```python3
picking_type = m.picking_id and m.picking_id.picking_type_id or m.picking_type_id
```

Reply:
Already tried something like that, still having CacheMiss exceptions...

First case when testing in test_move.py line 1986 in test_link_assign_7.

Preferred to keep notation coherent with already present ones and not to interfere with other processes.

Just debugged this morning, origin of CacheMiss is extra-moves (to be confirmed) created in _action_done, then merged in _action_confirm and unlinked in _merge_moves.

Do not know any efficient method to detect that move_to_confirm have been deleted...